crates/node/core/src/node_config.rs

```diff
@@ -422,6 +422,14 @@ impl<ChainSpec> NodeConfig<ChainSpec> {
             Err(e) => Err(eyre!("Failed to load configuration: {e}")),
         }
     }
+
+    /// Maps the config using the provided closure to update the [`ChainSpec`] generic.
+    pub fn map_chainspec<F, C>(self, f: F) -> NodeConfig<C>
+    where
+        F: FnOnce(Self) -> NodeConfig<C>,
+    {
+        f(self)
```

Comment:
ah okay, sorry I gave somewhat conflicting suggestions here, imo we can limit this to chainspec only so the closure should actually return the chainspec type given the current chainspec as input so probably `FnOnce(ChainSpec) -> C`

then we can move all of this in this function
```
NodeConfig {
                chain,
                datadir: config.datadir,
                config: config.config,
                metrics: config.metrics,
                instance: config.instance,
                network: config.network,
                rpc: config.rpc,
                txpool: config.txpool,
                builder: config.builder,
                debug: config.debug,
                db: config.db,
                dev: config.dev,
                pruning: config.pruning,
            }
```

Reply:
ah yeah I wasn't sure based on the pseudo code but this makes more sense haha